test/properties.ts

```diff
@@ -119,4 +122,52 @@ test('configured options', async (t) => {
     });
 
     return lock.latch;
+});
+
+
+
+test('noEcho', async(t) => {
+    t.plan(1);
+    let lock = new Lock();
+    let sc = t.context as SC;
+    let subj = createInbox();
+    let cp = connect({url: sc.server.nats, noEcho: true});
+    cp.then(async (nc) => {
+        let c2 = 0;
+        let sub2 = nc.subscribe(subj, ()=> {
+            c2++;
+        });
+        nc.publish(subj);
+        await nc.flush();
+        t.is(c2, 0);
+        nc.close();
+        lock.unlock();
+    }).catch((err) => {
+        if(err.code === ErrorCode.NO_ECHO_NOT_SUPPORTED) {
+            t.pass();
+        } else {
+            t.fail(err);
+        }
+        lock.unlock();
+    });
+    await lock.latch;
+});
+
+
+test('noEcho not supported', async(t) => {
+    let lock = new Lock();
+    let server = new mockserver.ScriptedServer(0);
+    try {
+        await server.start();
+    } catch(ex) {
+        t.log(ex);
+    }
+    t.plan(1);
+    let nc = await connect({port: server.port, noEcho: true});
+    nc.on('error', (err) => {
+        t.is(err.code, ErrorCode.NO_ECHO_NOT_SUPPORTED);
```

Comment:
This check that if there is an error, it is the expected one. But I don't see how this test ensure that the connect actually fails. In other words, there should be something that says on connect success, test fails.

Reply:
We discussed and this is ok. In the implementation, the connect itself succeeds, but then the error "callback" is invoked. So this test makes sure that this is the case and so this is correct.